specification/dartLangSpec.tex

```diff
@@ -17671,10 +18063,10 @@ \subsubsection{For-in}
 
 \commentary{%
 It follows that it is a compile-time error
-if $D$ is empty and \id{} is a final variable;
-and it is a dynamic error if $e$ has a top type,
+if $D$ is empty and \id{} is an immutable variable;
```

Comment:
What if `id` is late and immutable, and hasn't been assigned yet, and the iterable has length at most one? Or just `final` and unassigned?

Reply:
See the next comment.